File: examples/tcp_example.py
#!/usr/bin/env python

from hurricane import Gateway, Message
from hurricane.erl_codec import SocketWrapper

def main():
    gateway = Gateway(SocketWrapper('localhost', 3000))
    while True:
        message = Message()
        message.type = 'request'
        message.destination = 'time_server'
        message.tag = 0
        message.data = None
        gateway.send(message)
        print gateway.recv()

if __name__ == '__main__':
    main()

File: examples/tcp_example.rb
#!/usr/bin/env ruby

require 'hurricane'

gateway = Hurricane::Gateway.new(
  Erlang::SocketWrapper.new('localhost', 3000))
loop do
  request = Hurricane::Message.new()
  request.type = 'request'
  request.destination = 'time_server'
  request.tag = 0
  request.data = nil

  gateway.do_send(request)
  puts gateway.do_recv().inspect()
end
